editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -4380,7 +4384,13 @@ void CanvasItemEditor::_button_tool_select(int p_index) {
 	}
 
 	tool = (Tool)p_index;
+
 	viewport->update();
+	_update_cursor();
+
+	// Request immediate refresh of cursor when using hot-keys to switch between tools
+	DisplayServer::CursorShape ds_cursor_shape = (DisplayServer::CursorShape)viewport->get_default_cursor_shape();
```

Comment:
Shouldn't this be done inside the update_cursor function ?

Reply:
I didn't want to alter the way how ```_gui_input_viewport``` works - so ```_update_cursor``` it's only simple extraction of cursor selection, and immediate request for refresh is only called in ```_button_tool_select``` then.

Reply:
OK, I guess it makes sense.